### DISCUSSION TITLE
The 'NoBuild' property was set to true but the 'Build' target was invoked.

### DISCUSSION THREAD

**Discussion by graphicsxp** [teroneko/Teronis.DotNet#11] [Q&A] [answered]
I am trying to use your package to include ref projects dlls to my nuget package. To build the package, the following command is executed : 

`dotnet pack  .\$(ProjectName).csproj  --include-symbols --no-build -o C:\dev\Packages`

I am getting the error message :  
The 'NoBuild' property was set to true but the 'Build' target was invoked.

If I remove the --no-build parameters then I don't get the error message but I'd like to keep it, otherwise the solution is rebuilt a second time as it is already built as a first step in my build pipeline.

Is this possible ?

**Comment by graphicsxp** [comment]
I think this happens because I am not generating package on build (GeneratePackageOnBuild project property set to false).

Instead I generate package in post build 
```

 <Target Name="NugetPack">
     
		<Message Importance="High" Text="dotnet pack" />
		<Exec Command="dotnet pack  .\$(ProjectName).csproj  --include-symbols --no-build -o C:\dev\UPP\" />
    </Target>

```
Error is triggered by following code in Microsof.NET.sdk.targets :
```

  <!-- TODO: this target should not check GeneratePackageOnBuild.
             remove when https://github.com/NuGet/Home/issues/7801 is fixed.
    -->
  <Target Name="_CheckForBuildWithNoBuild"
          Condition="'$(NoBuild)' == 'true' and '$(GeneratePackageOnBuild)' != 'true'">
    <NETSdkError ResourceName="NoBuildRequested" />
  </Target>
```

If I don't use your package, I don't get the error. Obviously my csproj ref dll are no longer included in the generated package, but at least there is no error, so I'm wondering why your package leads to this error ?

**Comment by teneko** [comment, marked as answer]
I tried to track down the problem. I cannot find the problem. What the program is actually doing is just injecting custom target in the the process of packing (compare https://github.com/teroneko/Teronis.DotNet/blob/9b206bb711670bc352d4dbf73249bc43fbed037e/src/MSBuild/Packaging/ProjectBuildInPackage/src/build/Teronis.MSBuild.Packaging.ProjectBuildInPackage.targets#L11 and https://github.com/NuGet/NuGet.Client/blob/ea0bf4a379b6306fd88ff1529dc57e9b16af326a/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L434). I do not build. the custom target CopyProjectBuildInPackage (see https://github.com/teroneko/Teronis.DotNet/blob/9b206bb711670bc352d4dbf73249bc43fbed037e/src/MSBuild/Packaging/ProjectBuildInPackage/src/build/Teronis.MSBuild.Packaging.ProjectBuildInPackage.targets#L16) just depends on the targets BuildOnlySettings and ResolveReferences (see https://github.com/dotnet/msbuild/blob/master/src/Tasks/Microsoft.Common.CurrentVersion.targets) but they normally are sub targets of the build itself.
This means, that `dotnet pack` is going to build your project but that's in conflict with your parameter `--no-build`.
I can remember that the program `dotnet pack` is more user friendly and builds your project before it actually packs it. It only builds your project only once unless you have made changes. If you really just want to "pack", then you have to call the pack target manually with `dotnet msbuild -t:pack` (see https://github.com/NuGet/NuGet.Client/blob/ea0bf4a379b6306fd88ff1529dc57e9b16af326a/src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets#L202).
